overlord/devicestate/devicestate_gadget_test.go

```diff
@@ -906,3 +908,186 @@ func (s *deviceMgrGadgetSuite) TestUpdateGadgetOnCoreOldIsInvalidNowButShouldWor
 `
 	s.testUpdateGadgetOnCoreSimple(c, "", encryption, hybridGadgetYamlBroken, hybridGadgetYaml)
 }
+
+func (s *deviceMgrGadgetSuite) setupKernelAssetsUpdate(c *C) (chg *state.Change, tsk *state.Task) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	siGadget := &snap.SideInfo{
+		RealName: "foo-gadget",
+		Revision: snap.R(1),
+		SnapID:   "foo-gadget-id",
+	}
+	gadgetSnapYaml := "name: foo-gadget\nversion: 1.0\ntype: gadget"
+	gadgetYamlContent := `
+volumes:
+  pi:
+    schema: mbr
+    bootloader: u-boot
+    structure:
+      - name: foo
+        filesystem: vfat
+        type: 0C
+        size: 1200M
+        content:
+          - source: boot-assets/
+            target: /
+          - source: kernel:ref/managed-asset
```

Comment:
shouldn't this be

```suggestion
          - source: $kernel:ref/managed-asset
```
?

Reply:
also it seems as the test is currently written this value doesn't actually need to be anything, I can delete this whole source/target combo from content and the tests still pass :thinking: do we need to check more things in the mocked GadgetUpdate?

Reply:
Yes, good catch. This helper was doing way too much for what this meant to test. Sorry for this. I simplified (and renamed) it now and it should now really only do the minimal amount that is needed to trigger the `gadget.Update()` code with the right parameters.